app/components/Footer.tsx

```diff
@@ -1,7 +1,93 @@
+"use client";
+
+import Link from "next/link";
+import { usePathname } from "next/navigation";
+import { Github, Linkedin } from "lucide-react";
+
+type FooterNavLinkProps = {
+  href: string;
+  text: string;
+};
+
+const FooterNavLink = ({ href, text }: FooterNavLinkProps) => {
+  const currentRoute = usePathname();
+
+  const shouldHaveUnderline =
+    currentRoute.includes(href.slice(1)) ||
+    (currentRoute === "/" && href === "/#about");
+
+  return (
+    <Link
+      href={href}
+      className="text-neutral-500 hover:text-neutral-100 w-fit whitespace-nowrap transition-all duration-300"
+    >
+      <span>{text}</span>
+      <div
+        className={`w-6 h-[1px] ${
```

Comment:
I think this needs a little tweaking. When I tried this on my system (Chrome) the underline was only about 25% of the text regardless of how long the word was. I think this probably should be the entire word? Otherwise it feels incomplete to me...

![image](https://github.com/Barnstorm-Research/brc-web/assets/13735131/c4bf6cd8-c13b-401e-b608-5afe2e70f523)